runtime/python/pose_estimation/pose_estimation.py

```diff
@@ -0,0 +1,247 @@
+        nms_iou_thresh (float): IoU threshold for NMS.
+        regression_length (int): Maximum regression value for bounding boxes.
+        strides (list[int]): Stride values for each prediction scale.
+    """
+    image_id = 0
+    while True:
+        result = output_queue.get()
+        if result is None:
+            break  # Exit the loop if sentinel value is received
+
+        processed_image, raw_detections = result
+        post_process_and_save_inference_results(
+            processed_image, raw_detections, output_path, image_id,
+            height, width, class_num,
+            max_detections, score_threshold, nms_iou_thresh, regression_length, strides
+        )
+
+        image_id += 1
+
+
+def infer(
+    images: list[Image.Image],
+    net_path: str,
+    batch_size: int,
+    class_num: int,
+    output_path: Path,
+    data_type_dict: dict,
+    max_detections: int,
+    score_threshold: float,
+    nms_iou_thresh: float,
+    regression_length: int,
+    strides: list[int]
+) -> None:
+    """
+    Run inference with HailoAsyncInference, handle processes, and ensure proper cleanup.
+
+    Args:
+        images (list[Image.Image]): List of images to process.
+        net_path (str): Path to the HEF model file.
+        batch_size (int): Number of images per batch.
+        class_num (int): Number of classes.
+        output_path (Path): Path to save the output images.
+        data_type_dict (dict): Dictionary of layer names and data types.
+        max_detections (int): Maximal number of detections per class.
+        score_threshold (float): Confidence threshold for filtering.
+        nms_iou_thresh (float): IoU threshold for NMS.
+        regression_length (int): Maximum regression value for bounding boxes.
+        strides (list[int]): Stride values for each prediction scale.
+    """
+    input_queue = mp.Queue()
+    output_queue = mp.Queue()
+
+    hailo_inference = HailoAsyncInference(
+        net_path, input_queue, output_queue, batch_size, output_type=data_type_dict
+    )
+    height, width, _ = hailo_inference.get_input_shape()
+
+    enqueuing_process = Process(
+        target=enqueue_images,
+        args=(images, batch_size, input_queue, width, height)
+    )
+    processing_process = Process(
+        target=process_output,
+        args=(
+            output_queue, output_path, width, height, class_num, 
+            max_detections, score_threshold, nms_iou_thresh, regression_length, strides
+        )
+    )
+
+    enqueuing_process.start()
+    processing_process.start()
+
+    try:
+        hailo_inference.run()
+        enqueuing_process.join()
+        # To signal processing process to exit
+        output_queue.put(None)
+        processing_process.join()
+      
+        # Check for any exceptions in the processes
+        if enqueuing_process.exitcode or processing_process.exitcode != 0:
+            raise RuntimeError("Process terminated with an error")
```

Comment:
nice

Reply:
maybe logging?

Reply:
added a function in pose_estimation_utils:

```
def check_process_errors(*processes: Process) -> None:
    """
    Check the exit codes of processes and log errors if any process has a non-zero exit code.

    Args:
        processes (Process): The processes to check.
    """
    process_failed = False
    for process in processes:
        if process.exitcode != 0:
            logging.error(f"{process.name} terminated with an error. Exit code: {process.exitcode}")
            process_failed = True
    if process_failed:
        raise RuntimeError("One or more processes terminated with an error.")
```

 that it being called inside the _try_ :
 ```
 image_processor.join()
 check_process_errors(image_enqueuer, image_processor)
 logging.info(f'Inference was successful! Results have been saved in {output_path}')
```